src/java/uk/ac/rdg/resc/edal/cdm/DataChunk.java

```diff
@@ -118,10 +118,11 @@ public Index getIndex() {
      * @return the data value, or {@link Float#NaN} if this is a missing value
      */
     public float readFloatValue(Index index) {
-        double val = arr.getDouble(index);
-        if (this.needsEnhance)
-            return (float) this.var.convertScaleOffsetMissing(val);
-        else
-            return (float) val;
+        double val = arr.getFloat(index);
```

Comment:
In this case, I copied the code exactly as it is in the ncWMS code. This is one of the many classes we copy "as is" to keep in sync with the Reading SVN code. I don't mind adding a comment, as it shouldn't tank the diffs by to much.